src/basic/missing_socket.h

```diff
@@ -1,6 +1,7 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+#include <netinet/in.h>
```

Comment:
@yuwata : I tried to explain this in the commit message, but that was too succinct. If I drop it, I have other issues, e.g.

```
[171/979] Compiling C object src/libsystemd/libsystemd_static.a.p/sd-device_device-monitor.c.o
FAILED: src/libsystemd/libsystemd_static.a.p/sd-device_device-monitor.c.o 
ccache cc -Isrc/libsystemd/libsystemd_static.a.p -Isrc/libsystemd -I../src/libsystemd -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wno-nonnull-compare -Warray-bounds -Warray-bounds=2 -Wdate-time -Wendif-labels -Werror=format=2 -Werror=format-signedness -Werror=implicit-function-declaration -Werror=implicit-int -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=missing-declarations -Werror=missing-prototypes -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=strict-flex-arrays -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wunused-function -Wwrite-strings -Wzero-length-bounds -fdiagnostics-show-option -fno-common -fstack-protector -fstack-protector-strong -fstrict-flex-arrays --param=ssp-buffer-size=4 -Wno-unused-result -Werror=shadow -fPIC -fno-strict-aliasing -fstrict-flex-arrays=1 -fvisibility=hidden -fno-omit-frame-pointer -include config.h -pthread -fvisibility=default -MD -MQ src/libsystemd/libsystemd_static.a.p/sd-device_device-monitor.c.o -MF src/libsystemd/libsystemd_static.a.p/sd-device_device-monitor.c.o.d -o src/libsystemd/libsystemd_static.a.p/sd-device_device-monitor.c.o -c ../src/libsystemd/sd-device/device-monitor.c
In file included from ../src/libsystemd/sd-device/device-monitor.c:24:
../src/basic/missing_socket.h:102:23: error: expected identifier before numeric constant
  102 | #define IPPROTO_MPTCP 262
      |                       ^~~
```

And if I move the code to `missing_network.h` as you suggested in your other comment, I have:

```
[88/983] Compiling C object src/libsystemd-network/libsystemd-network.a.p/lldp-network.c.o
FAILED: src/libsystemd-network/libsystemd-network.a.p/lldp-network.c.o 
ccache cc -Isrc/libsystemd-network/libsystemd-network.a.p -Isrc/libsystemd-network -I../src/libsystemd-network -Isrc/basic -I../src/basic -Isrc/fundamental -I../src/fundamental -Isrc/systemd -I../src/systemd -I. -I.. -I../src/libsystemd/sd-bus -I../src/libsystemd/sd-device -I../src/libsystemd/sd-event -I../src/libsystemd/sd-hwdb -I../src/libsystemd/sd-id128 -I../src/libsystemd/sd-journal -I../src/libsystemd/sd-netlink -I../src/libsystemd/sd-network -I../src/libsystemd/sd-resolve -Isrc/shared -I../src/shared -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -std=gnu11 -O0 -g -Wno-missing-field-initializers -Wno-unused-parameter -Wno-nonnull-compare -Warray-bounds -Warray-bounds=2 -Wdate-time -Wendif-labels -Werror=format=2 -Werror=format-signedness -Werror=implicit-function-declaration -Werror=implicit-int -Werror=incompatible-pointer-types -Werror=int-conversion -Werror=missing-declarations -Werror=missing-prototypes -Werror=overflow -Werror=override-init -Werror=return-type -Werror=shift-count-overflow -Werror=shift-overflow=2 -Werror=strict-flex-arrays -Werror=undef -Wfloat-equal -Wimplicit-fallthrough=5 -Winit-self -Wlogical-op -Wmissing-include-dirs -Wmissing-noreturn -Wnested-externs -Wold-style-definition -Wpointer-arith -Wredundant-decls -Wshadow -Wstrict-aliasing=2 -Wstrict-prototypes -Wsuggest-attribute=noreturn -Wunused-function -Wwrite-strings -Wzero-length-bounds -fdiagnostics-show-option -fno-common -fstack-protector -fstack-protector-strong -fstrict-flex-arrays --param=ssp-buffer-size=4 -Wno-unused-result -Werror=shadow -fPIC -fno-strict-aliasing -fstrict-flex-arrays=1 -fvisibility=hidden -fno-omit-frame-pointer -include config.h -MD -MQ src/libsystemd-network/libsystemd-network.a.p/lldp-network.c.o -MF src/libsystemd-network/libsystemd-network.a.p/lldp-network.c.o.d -o src/libsystemd-network/libsystemd-network.a.p/lldp-network.c.o -c ../src/libsystemd-network/lldp-network.c
In file included from ../src/libsystemd-network/lldp-network.c:8:
../src/basic/missing_network.h:15:23: error: expected identifier before numeric constant
   15 | #define IPPROTO_MPTCP 262
      |                       ^~~
```

That's because these headers files are included directly by some .c files, before `<netinet/in.h>` or `socket-util.h`.

Should I fix the headers in the other files then? But it might be not very clear this file cannot be included directly, or before others, maybe better to make it self-contain?

Reply:
Including the netinet/in.h in missing_network.h is OK. I just meant that the header should be dropped from missing_socket.h.

Reply:
@yuwata: OK, thank you, this is what I did in the last version version.